src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -141,13 +195,121 @@ await ChangeNamespaceToMatchFoldersAsync(solutionAfterNamespaceChange, id, decla
             }
         }
 
-        private ImmutableArray<ISymbol> GetDeclaredSymbolsInContainer(
-            SemanticModel semanticModel,
-            SyntaxNode node,
+        protected async Task<ImmutableArray<(DocumentId, SyntaxNode)>> TryGetApplicableContainersFromAllDocumentsAsync(
+            Solution solution,
+            ImmutableArray<DocumentId> ids,
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            // If the node specified by span doesn't meet the requirement to be an applicable container in any of the documents 
+            // (See `TryGetApplicableContainerFromSpanAsync`), or we are getting different namespace declarations among 
+            // those documents, then we know we can't make a proper code change. We will return null and the check 
+            // will return false. We use span of namespace declaration found in each document to decide if they are identical.            
+
+            var documents = ids.SelectAsArray(id => solution.GetDocument(id));
+            var containers = ArrayBuilder<(DocumentId, SyntaxNode)>.GetInstance(ids.Length);
+            var spanForContainers = PooledHashSet<TextSpan>.GetInstance();
```

Comment:
Since we only care if a single TextSpan exists, why not use a nullable type here and just check for that?